### PR TITLE
Update pack variant from develop

### LINKED ISSUE #3107: Merge pack variant into development
* Update pack variants from development
* PR for full pack variants
* Check all UI areas, compare with develop (good time for writing external documentation, and test docs ?)

### PR DESCRIPTION
**Pull request by andreievg**
Fixes #3107

# 👩🏻‍💻 What does this PR do? 

This PR has all pack variant work up to date
* Hooks and UI elements
* Pack variant editing
* Sync refactor
* omSupply central sync server
* Sync documentation

# 🧪 How has/should this change been tested? 

Test suit to follow in this issue: https://github.com/msupply-foundation/open-msupply/issues/3109

Back end tests pass (including integration tests). And front end tests all pass but one (which also doesnt pass in develop)

## 💌 Any notes for the reviewer?

I did a pretty thorough self review. 

A few UI elements seem to be broken listed in this issue, will fix them up tomorrow in another PR

### CODE REVIEW

**Review by jmbrunskill — APPROVED** [review]
Obviously too big to review properly and lots of the question I have I'm sure have been discussed in other PRs such as why we need the `Delete`, `Upsert` traits for example.

Good to merge when your ready @andreievg

**Comment by andreievg** [comment]
Thanks for review @jmbrunskill.

I want to merge this one together with: https://github.com/msupply-foundation/open-msupply/pull/3127, since quite a few UI/UX tidy ups after merge and conflict resolution

**Pull request merged**
